utils/parser.js

```diff
@@ -0,0 +1,141 @@
+const controllers = require('../controllers');
+
+module.exports.qs = async url => {
+  const index = url.indexOf('?');
+
+  if (index === -1) return {};
+
+  const qs = url.substring(index + 1);
+  const queries = qs.split('&');
+
+  return queries.reduce((acc, curr) => {
+    const query = curr.split('=');
+    acc[query[0]] = query[1];
+    return acc;
+  }, {});
+};
+
+module.exports.genres = async ({ name, id }) => {
+  return {
+    name,
+    external_ids: {
+      tmdb_id: id,
+    },
+  };
+};
+
+module.exports.movie = async (data, configs) => {
+  return Object.assign(
+    {
+      adult: data.adult,
+      homepage: data.homepage,
+      original_language: data.original_language,
+      original_title: data.original_title,
+      overview: data.overview,
+      release_date: data.release_date,
+      runtime: data.runtime,
+      status: data.status,
+      tagline: data.tagline,
+      title: data.title,
+      external_ids: data.external_ids,
+    },
+    ...(await Promise.all([
+      parseMovieCredits(data.credits),
+      parseMovieGenres(data.genres),
+      parseMovieProductionCompanies(data.production_companies, configs),
+      parseMovieRatings(data.ratings),
+      parseMovieImages(data.images, configs),
+      parseMovieVideos(data.videos.results, configs),
+    ])),
+  );
+};
+
+const parseMovieCredits = async ({ cast = [], crew = [] }) => {
+  return {
+    credits: {
+      cast: cast.map(({ character, name, order }) => {
+        return { character, name, order };
+      }),
+      crew: crew.map(({ department, job, name }) => {
+        return { department, job, name };
+      }),
+    },
+  };
+};
+
+const parseMovieGenres = async (genres = []) => {
+  return {
+    genre_ids: (await Promise.all(
+      genres
+        .map(async genre => await controllers.genre.readOneByKey(genre.id))
+        .map(async req => await req),
+    )).map(res => res._id),
+  };
+};
+
+const parseMovieProductionCompanies = async (companies = [], configs) => {
```

Comment:
Should be implicit return

```suggestion
const parseMovieProductionCompanies = async (companies = [], configs) => ({
```